pin.sh

```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env bash
+# Set up flake registry to get o1js with all the submodules
+
+# Find the root of the o1js repo
+ROOT=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+pushd "$ROOT"
```

Comment:
what's the point of `pushd`/`popd` if we simply could do `cd "$ROOT"` without changing dir back (dir will be changed only within this script, not from above)

Reply:
I generally try to use pushd and popd in scripts cause they feel a bit easier to reason about, but I doesn't make a difference here and I don't feel strongly about it if you think cd is better I'm happy to change it.